### PR TITLE
update to new std.json api

### LINKED ISSUE #965: Better errors when parsing zls.json config file
### Zig Version

0.11.0-dev.1568+c9b957c93

### Zig Language Server Version

0.11.0-dev.160+384f227

### Steps to Reproduce

Using this configuration
```json
{
    "enable_snippets": true,
    "enable_ast_check_diagnostics": true,
    "enable_autofix": true,
    "enable_import_embedfile_argument_completions": true,
    "warn_style": true,
    "use_comptime_interpreter": true,
    "enable_semantic_tokens": false,
    "enable_inlay_hints": true,
    "operator_completions": true,
    "include_at_in_builtins": false,
}

```

open neovim with lsp-config installed.
Run :LspLog
you would have this output
```
[START][2023-02-04 16:51:49] LSP logging initiated
[ERROR][2023-02-04 16:51:49] .../vim/lsp/rpc.lua:734 "rpc" "zls" "stderr"
"warning: (config): Error while parsing configuration file: error.InvalidValueBegin
info : ( main ): No config file zls.json found.
info : (config): Using zig executable /usr/bin/zig
info : (config): Using zig lib path '/firefox/home/.local/zig/zig-linux-x86_64-0.11.0-dev.1568+c9b957c93/lib'
info : (server): client is 'Neovim-0.9.0'
info : (server): zls initializing
info : (server): Server.ClientCapabilities{ .supports_snippets = true, .supports_apply_edits = true, .supports_will_save = true, .supports_will_save_wait_until = true, .supports_publish_diagnostics = true, .supports_code_action_fixall = false, .hover_supports_md = true, .completion_doc_supports_md = true, .label_details_support = true, .supports_configuration = true }
info : (server): Using offset encoding: utf-16"
[ERROR][2023-02-04 16:52:01] .../vim/lsp/rpc.lua:734 "rpc" "zls" "stderr"
"info : (server): Setting configuration...
info : (config): Using zig executable /usr/bin/zig"
```

This behaviour is also experienced with 
`zls --show-config-path`
```
warning: (config): Error while parsing configuration file: error.InvalidValueBegin
/firefox/home/.config/zls.json
```

I tested this with both the nightly version of `Neovim-0.9.0` and `Neovim-0.8.3` on `archlinux`

### Expected Behavior

The `zls.json` configuration file is read by zls successfully and used. That is, there should be no error while parsing the configuration file and the configuration should be respected and override the default configurations.

### Actual Behavior

`zls --show-config-path`
```
warning: (config): Error while parsing configuration file: error.InvalidValueBegin
/firefox/home/.config/zls.json
```

and from nvim the above error and also 
`info : ( main ): No config file zls.json`

I noticed this because I realized some configuration options aren't applied and found these errors.

### LINKED ISSUE #1190: Zig master changed json API
### Zig Version

0.11.0-dev.3138+87de8212ad

### Zig Language Server Version

91645612e61118af89b23d6766d47532b473827d

### Steps to Reproduce

Compile ZLS

### Expected Behavior

Should compile

### Actual Behavior

Error:

```
src/main.zig:337:52: error: expected type 'mem.Allocator', found 'Config'
        defer std.json.parseFree(Config, new_config.config, .{ .allocator = allocator });
                                         ~~~~~~~~~~^~~~~~~
```

Zig master changed the JSON API, `parseFree` arguments changed (`allocator` no longer an option) and `TokenStream` removed.

### PR DESCRIPTION
**Pull request by Techatrix**
tres is being updated in https://github.com/ziglibs/tres/pull/12
zig-lsp-codegen is being updated in https://github.com/zigtools/zig-lsp-codegen/pull/8
fixes #1190 
fixes #965

### CODE REVIEW

**Comment by sagehane** [comment]
It's highly likely that this would break the `flake.nix` build.
Could it be tested by making sure that `nix flake update --commit-lock-file && nix-build` succeeds?

File: build.zig [hunk 1]
```diff
@@ -7,7 +7,7 @@ const zls_version = std.builtin.Version{ .major = 0, .minor = 11, .patch = 0 };
 pub fn build(b: *std.build.Builder) !void {
     comptime {
         const current_zig = builtin.zig_version;
-        const min_zig = std.SemanticVersion.parse("0.11.0-dev.3003+e1f5ad3cc") catch unreachable; // std.http: do -> wait, fix redirects
+        const min_zig = std.SemanticVersion.parse("0.11.0-dev.3134-018b743c7") catch unreachable; // std.http: do -> wait, fix redirects
```
Comment (kuon) [inline build.zig:10]: That should be a `+` and not a `-`.

Comment (nektro) [reply build.zig:10]: the comment should also be updated

**Review by SuperAuguste — APPROVED** [review]
lets gooo!!! thanks for your work on this @Techatrix - just fixed it up real quick and we're good to go 🚀

**Pull request merged**